pkg/dataset/sort_merge_join.go

```diff
@@ -77,8 +79,8 @@ func NewSortMergeJoin(joinType ast.JoinType, joinColumn *JoinColumn, outer proto
 	}
 
 	fields := make([]proto.Field, 0, len(outerFields)+len(innerFields))
-	fields = append(fields, outerFields...)
 	fields = append(fields, innerFields...)
```

Comment:
```go
fields := make([]proto.Field, len(innerFields)+len(outerFields))
copy(fields, innerFields)
copy(fields[len(innerFields):], outerFields)
```

Reply:
OK

Reply:
panic: runtime error: slice bounds out of range [3:0] [recovered]